apps/patient/src/views/Pharmacy.tsx

```diff
@@ -54,6 +81,142 @@ const query = (method, data) =>
     });
   });
 
+const geocode = async (address: string) => {
+  const data = await geocoder.geocode({ address });
+  if (data?.results) {
+    return {
+      address: data.results[0].formatted_address,
+      lat: data.results[0].geometry.location.lat(),
+      lng: data.results[0].geometry.location.lng()
+    };
+  } else {
+    throw new Error('Failed to geocode the address.');
+  }
+};
+
+const sortIndiePharmaciesFirst = (list: PharmacyType[], distance: number, limit: number) => {
+  const featuredPharmacies = list
+    .filter((p: PharmacyType) => AUSTIN_INDIE_PHARMACY_IDS.includes(p.id))
+    .slice(0, limit);
+  const otherPharmacies = list.filter(
+    (p: PharmacyType) => !featuredPharmacies.some((f) => f.id === p.id)
+  );
+  return featuredPharmacies.concat(otherPharmacies);
+};
+
+const getPharmacies = async (
+  searchParams: {
+    latitude: number;
+    longitude: number;
+    radius: number;
+  },
+  limit: number,
+  offset: number,
+  token: string
+) => {
+  graphQLClient.setHeader('x-photon-auth', token);
+
+  try {
+    const query: { pharmaciesByLocation: types.Pharmacy[] } = await graphQLClient.request(
+      GET_PHARMACIES,
+      {
+        location: searchParams,
+        limit,
+        offset
+      }
+    );
+    return query.pharmaciesByLocation;
+  } catch (error) {
+    if (
+      error?.response?.errors &&
+      error.response.errors[0].message === 'No pharmacies found near location'
+    ) {
+      const { message } = error.response.errors[0];
+      console.log(message);
+      return message;
```

Comment:
Don't like taht this function sometimes returns a list of pharmacies, sometimes a string, and sometimes an error

Reply:
ah updated